internal/pkg/poster/poster.go

```diff
@@ -70,154 +80,159 @@ func (p *Poster) StopPoSting() {
 }
 
 func (p *Poster) startPoStIfNeeded(ctx context.Context, newHead block.TipSet) error {
-	// TODO: Window PoSt (#4010)
-	//p.postMutex.Lock()
-	//defer p.postMutex.Unlock()
-	//
-	//if p.postCancel != nil {
-	//	// already posting
-	//	return nil
-	//}
-	//
-	//tipsetHeight, err := newHead.Height()
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//root, err := p.chain.GetTipSetStateRoot(ctx, newHead.Key())
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//stateView := p.stateViewer.StateView(root)
-	//provingPeriodStart, _, _, err := stateView.MinerProvingPeriod(ctx, p.minerAddr)
-	//if err != nil {
-	//	return nil
-	//}
-	//
-	//if provingPeriodStart > tipsetHeight {
-	//	// it's not time to PoSt
-	//	return nil
-	//}
-	//
-	//ctx, p.postCancel = context.WithCancel(ctx)
-	//go p.doPoSt(ctx, stateView, provingPeriodStart, newHead.Key())
-	//
+	p.postMutex.Lock()
+	defer p.postMutex.Unlock()
+
+	if p.postCancel != nil {
```

Comment:
Where does this get reset to nil?

Reply:
We call `p.cancelPost()` instead of `p.postCancel()`. It gets set to nil in that method.